### PR TITLE
add support for custom post-install scripts

### PR DESCRIPTION
**Pull request by AGhost-7**
# Description
Adds a script hook to define custom scripts after the installation completes. This allows me to copy from a ConfigMap in kubernetes the configuration files for valheim plus et al. Example post-install script:

```
#!/usr/bin/env bash
set -ex

rm -rf /home/steam/valheim/BepInEx/config
cp -r /home/steam/bep-config /home/steam/valheim/BepInEx/config
chown -R 1000:1000 /home/steam/valheim/BepInEx/config
```

Where `/home/steam/bep-config` is a volume in kubernetes pointing to a ConfigMap.

## Contributions

- I changed the startup script because I couldn't attach extra logic between the mod installation and server startup.
- Added logic executing all scripts under `/valheim-post-install.d/` before the valheim server starts.

## Checklist

- [ ] I added one or multiple labels which best describes this PR.
- [x] I have tested the changes locally.
- [ ] This PR has a reviewer on it.
- [x] I have validated my changes in a docker container and on Ubuntu. (Only needed for Odin or Docker Changes)

### CODE REVIEW

**Comment by mbround18** [comment]
Are these to be executed as root or the steam user?

**Comment by AGhost-7** [comment]
For my use case being executed as the steam user worked fine. I'm also using this to install additional plugins such as epic loot which doesn't seem to install correctly due to its non-standard structure.

File: src/scripts/start_valheim.sh [hunk 1]
```diff
@@ -122,6 +122,11 @@ if [ ! "${TYPE}" = "vanilla" ]; then
   done
 fi
 
+if [ -d /start-valheim-post-install.d/ ]; then
```
Comment (mbround18) [inline src/scripts/start_valheim.sh:125]: Mind of we change up the dir name and add quotes? 

Quotes for standardization and dir name for simplification. 

```bash
if [ -d "/valheim-post-install.d/" ]; then
  log "Executing post-install scripts"
  find /valheim-post-install.d/ -type f -executable -exec {} \;
fi
```

**Review by mbround18 — CHANGES_REQUESTED** [review]

**Comment by mbround18** [comment]
I do have a question, unrelated to this PR, why not mount your config map to `/home/steam/valheim/BepInEx/config` ?

**Comment by AGhost-7** [comment]
There's already a mount in `/home/steam/valheim`, wasn't working for me on k3s.

**Pull request merged**